components/browser/menu/src/main/java/mozilla/components/browser/menu/view/StickyFooterLinearLayoutManager.kt

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu.view
+
+import android.content.Context
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+
+/**
+ * Vertical LinearLayoutManager that will ensure an item at a position specified through
+ * [StickyItemsAdapter.isStickyItem] will not scroll past the list bottom.
+ *
+ * The list would otherwise scroll normally with the other elements being scrolled beneath the sticky item.
+ */
+open class StickyFooterLinearLayoutManager<T> constructor(
+    context: Context,
+    reverseLayout: Boolean = false
```

Comment:
We have a sticky footer **and** a sticky header layout manager. Do we also need `reverseLayout`? What is the use case for this?

Reply:
`reverseLayout` is a constructor parameter of Android's `LinearLayoutManager`, one that AC clients might expect to be able to change to control whether the list should be layout from top to bottom or from bottom to top - effectively change the visible item positions - that same menu item might appear visually at the top of the list or at the bottom.
The list can be configured with this property without also having an item set as sticky.
Then sticky header/footer just prevents a specific item from being scrolled off to the top / bottom of the list.